### ISSUE TITLE
[GraphQL] : Null values in Category object are breaking the CategoryViewStrategy

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#21487]
Reported via Support: [106149](https://dotcms.zendesk.com/agent/tickets/106149)
Reproducible as of dotCMS 21.06.4, up to latest master (21.12)

When you query for the `_map` object in a GraphQL query and one of the returned contentlets has a **Category** field with values in it, the call will fail if there are properties in such a Category that are set as `null` instead of an empty String.

**To Reproduce**
Reach out to me for the appropriate data set to test this, and the STR. Here's an example of the stack trace generated under these circumstances:
```
[04/01/22 18:07:25:226 UTC] ERROR datafetcher.ContentMapDataFetcher: null value in entry: keywords=null
java.lang.NullPointerException: null value in entry: keywords=null
at com.google.common.collect.CollectPreconditions.checkEntryNotNull(CollectPreconditions.java:32) ~[guava-27.0.1-android.jar:?]
at com.google.common.collect.ImmutableMap$Builder.put(ImmutableMap.java:249) ~[guava-27.0.1-android.jar:?]
at com.dotmarketing.portlets.contentlet.transform.strategy.CategoryViewStrategy.transform(CategoryViewStrategy.java:104) ~[dotcms_21.06.4_999999.jar:?]
at com.dotmarketing.portlets.contentlet.transform.strategy.CategoryViewStrategy.transform(CategoryViewStrategy.java:86) ~[dotcms_21.06.4_999999.jar:?]
at com.dotmarketing.portlets.contentlet.transform.strategy.CategoryViewStrategy.transform(CategoryViewStrategy.java:61) ~[dotcms_21.06.4_999999.jar:?]
at com.dotmarketing.portlets.contentlet.transform.strategy.AbstractTransformStrategy.apply(AbstractTransformStrategy.java:95) ~[dotcms_21.06.4_999999.jar:?]
at com.dotmarketing.portlets.contentlet.transform.DotContentletTransformerImpl.lambda$3(DotContentletTransformerImpl.java:102) ~[dotcms_21.06.4_999999.jar:?]
at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
at java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720) ~[?:?]
...
...
...
```

**Expected behavior**
Our code needs to handle cases in which properties can be `null` or empty Strings, and still be valid.

**Additional context**
The currently valid workaround is to update the database directly, and replace the `NULL` values with an empty String.

**Comment by fabrizzio-dotCMS** [comment]
This is the GraphQL query that fire the exception
```
query PAGES2 {

page(

url: "/blog/options-trading"

pageMode: "live"

languageId: "1"

) {

title

urlContentMap {

_map

}

}

}
```

**Comment by bryanboza** [comment]
Fixed, tested on release-22.02 // Docker // Insomnia. After follow the provided steps I'm unable to get the error.
